NAMESPACE

```diff
@@ -1,6 +1,33 @@
 # Generated by roxygen2: do not edit by hand
 
-export(hash_salt)
-export(hash_seed)
-importFrom(Rcpp,evalCpp)
-useDynLib(hashprng, .registration = TRUE)
+export(r2dtable)
+export(rWishart)
+export(rbeta)
+export(rbinom)
+export(rcauchy)
+export(rchisq)
+export(rexp)
```

Comment:
Do you have an issue to approach not needing to do this? It seems like quite an unsafe design and one that I don't think will fly with CRAN

Reply:
I'm relatively confident this is actually what users want - minimal modification to their code to use this capability.

Reply:
Great but I don't think you will be able to get it on CRAN!

Reply:
:shrug: give the people what they want.

that said: i agree there are tradeoffs here. my goal: enable the user to switch matching on and off, without having to change code. this approach currently only makes the refactor easy-ish. so:
 - make the refactor easier: provide a bit of static analysis to find invocations of the overriden functions, that don't have a `hash = ...` argument
 - make the switching (back to base) possible with an option / package environment variable
 - be a bit noisier when loading the package

i think adding those also helps allay some of the overloading concerns. I think there's one other, harder problem to solve: making it possible to use hashprng within another package (that does some thing that requires matching) without having that leak out of that package.

Reply:
I think as suggested elsewhere if going to do this I wouldn't do it via reexporting but instead have a overload function that users can use and then use this is an option etc to overload a list of common r functions (i.e these)

Reply:
you'll still have to assign to the global environment which isn't great but at least its a bit more in line with normal practice?

Reply:
hmm, i don't recall that suggestion. makes some sense - so basically, users would do:

```r
library(hashprng)
hashoverload() # defaults to everything hashprng knows about
... some code ...
hashunload() # turns off overloads
# or hashoverload(c("unif", "binom")) # runif and rbinom only
# or hashoverload(c("custom")) # wraps some rcustom that it finds in the environment
```

that's a little more onerous, but definitely less likely to annoy the CRAN gods.

Reply:
still doesn't deal well with people having re-written their code to have `hash = ...` arguments, that need to be ignored if they've switched off matching.

Reply:
I think for the latter I would enforce they actually use a wrapper around their dists (that can have args the dists don't have) and then they can turn hashing on and off using an option.

It means they have to rewrite their code but they can easily the turn it on and off if they are so minded?

Reply:
one aside: R CMD check passes just fine.